### ISSUE TITLE
Suggestion for MorphIO concerning single point sections

### ISSUE THREAD

**Issue by eleftherioszisis** [BlueBrain/NeuroM#675]
Currently, the single point sections are discarded in the neurolucida loader:

https://github.com/BlueBrain/NeuroM/blob/10b10751e288328495e14c6185dd08d04224f1d3/neurom/io/neurolucida.py#L193-L198

I haven't checked if this is the case for h5 files as well. This may be a sensible treatment for the latter if duplicate points are present. In the case of a fork where one of the sections only consists of the duplicate point ( that of the parent), then there isn't any new information, and thus this branching point is redundant.

But, there is not such a standard for Neurolucida files, which may (in the case of bbp asc files for NEURON) or may not contain duplicate points. If there are no duplicate points then a single point section carries topological information for the branching structure of the tree albeit minimal.

My suggestion would be to load these cases unaltered and allow for optional filters that could for example remove these points if they are an issue in the statistical analysis, or in the case of topological analysis interpolate this point into a proper section by adding at least one more point.

In other words, morphio should avoid assumptions on the topology of the morphologies, provided it is a sensible one.

**Comment by wizmer** [comment]
In MorphIO when the point of the single point section is the duplicate point, the section is discarded.
See [here](https://github.com/BlueBrain/MorphIO/blob/5e111f3141f7a1ee72e0260111ce569741d80acb/tests/test_neurolucida.py#L350) and [here for a more complex case](https://github.com/BlueBrain/MorphIO/blob/5e111f3141f7a1ee72e0260111ce569741d80acb/tests/test_neurolucida.py#L374)

If the single point is not the duplicate point, then the duplicate is added and the section is perfectly valid. See [here](https://github.com/BlueBrain/MorphIO/blob/5e111f3141f7a1ee72e0260111ce569741d80acb/tests/test_neurolucida.py#L253)

**Comment by eleftherioszisis** [comment]
I see, it makes perfect sense.
